integration/evaluation_test.go

```diff
@@ -0,0 +1,658 @@
+package integration_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"testing"
+
+	structpb "github.com/golang/protobuf/ptypes/struct"
+
+	flagd "github.com/open-feature/go-sdk-contrib/providers/flagd/pkg"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+
+	"github.com/cucumber/godog"
+)
+
+// ctxStorageKey is the key used to pass test data across context.Context
+type ctxStorageKey struct{}
+
+var (
+	client *openfeature.Client
+)
+
+func init() {
+	// register the flagd provider before the tests
+	fmt.Println("Setting flagd provider...")
+	openfeature.SetProvider(flagd.NewProvider(flagd.WithPort(8013)))
+	fmt.Println("flagd provider configured!")
+
+	client = openfeature.NewClient("integration tests")
+}
+
+func aBooleanFlagWithKeyIsEvaluatedWithDefaultValue(
+	ctx context.Context, flagKey, defaultValueStr string,
+) (context.Context, error) {
+	defaultValue, err := strconv.ParseBool(defaultValueStr)
+	if err != nil {
+		return ctx, errors.New("default value must be of type bool")
+	}
+
+	got, err := client.BooleanValue(ctx, flagKey, defaultValue, openfeature.EvaluationContext{})
+	if err != nil {
+		return ctx, fmt.Errorf("openfeature client: %w", err)
+	}
+
+	return context.WithValue(ctx, ctxStorageKey{}, got), nil
+}
+
+func theResolvedBooleanValueShouldBe(ctx context.Context, expectedValueStr string) error {
+	expectedValue, err := strconv.ParseBool(expectedValueStr)
+	if err != nil {
+		return errors.New("expected value must be of type bool")
+	}
+
+	got, ok := ctx.Value(ctxStorageKey{}).(bool)
+	if !ok {
+		return errors.New("no flag resolution result")
+	}
+
+	if got != expectedValue {
+		return fmt.Errorf("expected resolved boolean value to be %t, got %t", expectedValue, got)
+	}
+
+	return nil
+}
+
+func aStringFlagWithKeyIsEvaluatedWithDefaultValue(
+	ctx context.Context, flagKey, defaultValue string,
+) (context.Context, error) {
+	got, err := client.StringValue(ctx, flagKey, defaultValue, openfeature.EvaluationContext{})
+	if err != nil {
+		return ctx, fmt.Errorf("openfeature client: %w", err)
+	}
+
+	return context.WithValue(ctx, ctxStorageKey{}, got), nil
+}
+
+func theResolvedStringValueShouldBe(ctx context.Context, expectedValue string) error {
```

Comment:
There's a few functions here like this one and `theResolvedIntegerValueShouldBe` that might be able to be made generic so they can work for multiple gherkin tests. Probably not possible with the boolean version wherein you'll need to parse.

This is entirely not-necessary though.

Reply:
Do you mean locating any functions where behaviour is being duplicated and merging them into a singular function?
I can't see any instances where this is happening. `theResolvedStringValueShouldBe` and `theResolvedStringResponseShouldBe` are similar in what they're asserting but the execution is different due to the way the data is store/retrieved.

Maybe I've misunderstood.

Reply:
I mean functions like `theResolvedStringValueShouldBe` and `theResolvedIntegerValueShouldBe`. These functions are the same, except the interface casting and string format token. which are associated with the type of the `expectedValue`. That's why I think (possibly incorrectly) they could be made generic.

This is absolutely not a necessary change though, even if I'm correct.

Reply:
I think due to the way we're storing the typed data and retrieving it for the assertion the complexity of doing this wouldn't be worth the benefits